src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -35,6 +47,207 @@ public ProjectView()
+
+            IEnumerable<NuGetVersion> versions = model.Versions.Where(v => v != null).Select(v => v.Version);
+            bool userTypedAVersionRange = ComboboxText.StartsWith("(", StringComparison.OrdinalIgnoreCase) || ComboboxText.StartsWith("[", StringComparison.OrdinalIgnoreCase);
+
+            switch (e.Key)
+            {
+                case Key.Enter:
+                    bool isValidNuGetVersionRange = _versions.SelectedIndex >= 0 ? VersionRange.TryParse(_versions.Items[_versions.SelectedIndex].ToString(), out VersionRange requestedVersionRange) : false;
+                    bool isUserInputValidNuGetVersionRange = _versions.SelectedIndex >= 0 ? VersionRange.TryParse(ComboboxText, out userRequestedVersionRange) : false;
+
+                    if (_versions.SelectedIndex >= 0 && isValidNuGetVersionRange)
+                    {
+                        if (userTypedAVersionRange)
+                        {
+                            NuGetVersion rangeBestVersion = userRequestedVersionRange.FindBestMatch(versions);
+                            bool isBestOption = rangeBestVersion.ToString() == _versions.Items[_versions.SelectedIndex].ToString();
+                            if (isBestOption)
+                            {
+                                _versions.SelectedValue = new DisplayVersion(userRequestedVersionRange, rangeBestVersion, additionalInfo: null);
+                                _versions.Text = ComboboxText;
+                                EditableTextBox.SelectionStart = 0;
+                                EditableTextBox.SelectionLength = int.MaxValue;
+                                _versions.IsDropDownOpen = false;
+
+                                e.Handled = true;
+                                break;
+                            }
+                        }
+
+                        _versions.SelectedValue = _versions.Items[_versions.SelectedIndex];
+                        _versions.Text = _versions.SelectedValue.ToString();
+                        EditableTextBox.SelectionStart = 0;
+                        EditableTextBox.SelectionLength = int.MaxValue;
+                    }
+
+                    _versions.IsDropDownOpen = false;
+                    e.Handled = true;
+                    break;
+                case Key.Down:
+                    if (_versions.SelectedIndex < _versions.Items.Count - 1)
+                    {
+                        // The separator is a null
+                        if (_versions.SelectedItem == null)
+                        {
+                            _versions.SelectedIndex++;
+                        }
+                        e.Handled = true;
+                    }
+                    break;
+                case Key.Up:
+                    if (_versions.SelectedIndex >= 0)
+                    {
+                        // The separator is a null
+                        if (_versions.SelectedItem == null)
```

Comment:
First thought: can we detect this earlier and do less work? eg, not needing to iterate over `model.versions` above, etc?

Second thought: I wish we could avoid having a null item for a separator. I looked into this a while back, and it seemed this is the only real way to show a separator in a combobox. I don't like that answer, so if you have the opportunity to look into it further, I'd like to learn about any alternatives you may find.